x/babylon/keeper/wasm.go

```diff
@@ -138,13 +19,21 @@ func (k Keeper) SendBeginBlockMsg(c context.Context) error {
 	ctx := sdk.UnwrapSDKContext(c)
 	headerInfo := ctx.HeaderInfo()
 
-	stakingAddr := k.getBTCStakingContractAddr(ctx)
-	finalityAddr := k.getBTCFinalityContractAddr(ctx)
-	if stakingAddr == nil || finalityAddr == nil {
+	contracts := k.GetBSNContracts(ctx)
+	if contracts == nil || !contracts.IsSet() {
 		k.Logger(ctx).Info("Skipping begin block processing: contract addresses are missing")
 		return nil
 	}
 
+	stakingAddr, err := sdk.AccAddressFromBech32(contracts.BtcStakingContract)
```

Comment:
Can we have `contracts.GetBtcStakingContractAddress()`? Same for other contracts